### ISSUE TITLE
"master" in cluster spec for distributed training 

### ISSUE THREAD

**Issue by llidev** [tensorflow/tensorflow#22182]
Hi,

Given in the sample code [mnist_replica.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/dist_test/python/mnist_replica.py), `tf.train.ClusterSpec()` needs to be used to define the cluster for distributed training.  The function `tf.train.ClusterSpec()` can only take `"ps"` and `"worker"` hosts, but `"master"` is not allowed. As far as I know, `"master"` is usually defined as a special worker and should be a part of the worker hosts, referred as `task_id = 0`. This makes sense to me.

However, in the context of `tf.estimator`, if `"cluster"` is set in `TF_CONFIG`, it must have one `"chief"` node, in addition to `"worker"`. For example,

```
TF_CONFIG='{
    "cluster": {
        "chief": ["host0:2222"],
        "worker": ["host1:2222", "host2:2222", "host3:2222"],
        "ps": ["host4:2222", "host5:2222"]
    },
    "task": {"type": "chief", "index": 0}
}'
```
I am trying to use `TF_CONFIG` for all our distributed training work, but it looks like to be inconsistent. For example, we need to do some preprocessing in the above `TF_CONFIG` (i.e., combining `"chief"` and `"worker"` hosts, and re-index them), to use it for `tf.train.ClusterSpec()`. 

Is there any better standard we can follow to achieve our goal?

Thanks

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by lliimsft** [comment]
Host OS: Ubuntu16.04LST
Docker image: 1.10.0-gpu

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sushreebarsa** [comment]
@llidev We see that you are using TF v1.x which is not actively supported and we recommend you to upgrade latest TF versions(2.4 or later).Please refer to the [migration](https://www.tensorflow.org/guide/migrate) doc .
Estimators are not recommended for new code. Estimators  can behave unexpectedly, especially when combined with TF 2 code. Could you please refer to this [link](https://www.tensorflow.org/api_docs/python/tf/estimator/DNNClassifier) for more details  and let us know if it helps?
Thanks!
